### ISSUE TITLE
How can I work with select option in a grid row?

### ISSUE THREAD

**Issue by abernardobr** [Knockout-Contrib/KoGrid#106]
I have successfully set up a cell on a row to be a select option using the template bellow:

<script type="text/html" id="kotemplate_select">
    <div data-bind="kgCell: $cell">
        <select data-mini="true" data-bind="options: $cellValue, event : { change : function(data, event) { mymodel.gridCallBack(data, event,  $cell, $cellValue); }  } ">
            <option value="undefined">Selecione...</option>
        </select>
    </div>
</script>

But I have not been able to find out the value of the select box. Is there a way I can set things so that I am able to know the value binding of the select per row for the specific cell?

Congrats for the great product.

Augusto

**Comment by orneryd** [comment]
currently there is no selected cell feature. this is something we can look at adding in the future.

**Comment by abernardobr** [comment]
Thanks for your response Timothy.

Here is what I have done to solve this issue:

 // GRID OVERRRIDES

kg.templates.generateRowTemplate = function (options) {

```
var b = new kg.utils.StringBuilder(),

    cols = options.columns;


 b.append('<div data-bind="kgRow: $data, click: $data.toggleSelected,
```

css: { \'kgSelected\': $data.selected }">');

```
 kg.utils.forEach(cols, function (col, i) {


     // check for the Selection Column

    if (col.field === '__kg_selected__') {

        b.append('<div class="kgSelectionCell" data-bind="kgCell: {
```

value: \'{0}\' } ">', col.field);

```
        b.append('  <input type="checkbox" data-bind="checked:
```

$data.selected" />');

```
        b.append('</div>');

    }

    // check for RowIndex Column

    else if (col.field === 'rowIndex') {

        // TODO: Change High Design

        // b.append('<div class="kgRowIndexCell" data-bind="kgCell: {
```

value: \'{0}\' } "></div>', col.field);

```
        b.append('<div class="kgRowIndexCell"></div>');

    }

    // check for a Column with a Cell Template

    else if (col.hasCellTemplate) {

        // first pull the template

        var tmpl = kg.templateManager.getTemplateText(col.cellTemplate);


         // build the replacement text

        var replacer = "{ value: '" + col.field + "' }";


         // TODO: Change High Design

        tmpl = tmpl.replace(/\$cellFieldValue/g, "$data." + col.field +
```

"Value");

```
        // run any changes on the template for re-usable templates

        tmpl = tmpl.replace(/\$cellClass/g, col.cellClass || 'kgEmpty');

        tmpl = tmpl.replace(/\$cellValue/g, "$data." + col.field);

        tmpl = tmpl.replace(/\$cell/g, replacer);


         b.append(tmpl);

    }

    // finally just use a basic template for the cell

    else {

        b.append('  <div class="{0}"  data-bind="kgCell: { value:
```

\'{1}\' } "></div>', col.cellClass || 'kgEmpty',  col.field);

```
    }

});


 b.append('</div>');


 return b.toString();
```

};

I have overridden the row template and created a cell value.

We are also working in being possible to "pin" a column. That is, have it
frozen on the left, while the rest of the grid can vertical scroll. If you
can point us to the right places to go, it would be great.

Regards,

Augusto

**Comment by orneryd** [comment]
Fixed in 2.0
